lang/cs/Org.Apache.REEF.Evaluator.Tests/TaskRuntimeTests.cs

```diff
@@ -214,6 +215,92 @@ public void TestTaskEventsOnFailure()
             Assert.Equal(testTaskEventStopHandler.StopInvoked.Value, taskId);
         }
 
+        /// <summary>
+        /// Tests that suspend ends the task and invokes the right handler.
+        /// </summary>
+        [Fact]
+        public void TestSuspendTask()
+        {
+            var contextId = Guid.NewGuid().ToString();
+            var taskId = Guid.NewGuid().ToString();
+
+            var injector = GetInjector(typeof(CountDownAction), contextId, taskId);
+            var taskRuntime = injector.GetInstance<TaskRuntime>();
+            taskRuntime.RunTask();
+
+            var task = injector.GetInstance<TestTask>();
+            taskRuntime.Suspend(null);
+
+            task.FinishCountdownEvent.Wait();
+            task.DisposeCountdownEvent.Wait();
+
+            Assert.True(task.SuspendInvoked);
+        }
+
+        /// <summary>
+        /// Tests that suspend is not invoked after task is done.
+        /// </summary>
+        [Fact]
+        public void TestSuspendTaskAfterDoneIsNotSuspended()
+        {
+            var contextId = Guid.NewGuid().ToString();
+            var taskId = Guid.NewGuid().ToString();
+
+            var injector = GetInjector(contextId, taskId);
+            var taskRuntime = injector.GetInstance<TaskRuntime>();
```

Comment:
Similar at here, bind TestTask to TaskConfigurationOptions.SuspendHandler explicitly would make the future injection clear for TaskRuntime as an example.